src/components/ListItem.js

```diff
@@ -0,0 +1,46 @@
+import { doc, updateDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import { useEffect, useState } from 'react';
+
+const oneDayInSeconds = 86400;
+
+const style = {
+       listStyleType: 'none',
+       textAlign: 'left'
+}
+const ListItem = ({ itemData }) => {
+    const [checked, setChecked] = useState(itemData.lastPurchased !== null)
+
+    const wasPurchasedWithin24Hours = () => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+        return (Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) <= oneDayInSeconds
```

Comment:
`Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds` 
This is used in a few places, maybe it can be put in a function so you avoid duplication :slightly_smiling_face: ?